instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts

```diff
@@ -9,6 +9,18 @@ group = "io.opentelemetry.instrumentation"
 val versions: Map<String, String> by project
 val springBootVersion = versions["org.springframework.boot"]
 
+// R2DBC is shadowed to prevent org.springframework.boot.autoconfigure.r2dbc.R2dbcAutoConfiguration
+// from being loaded by Spring Boot - even if the user doesn't want to use R2DBC.
```

Comment:
Could you please explain more the problem?

Reply:
yes I think you should explain that spring `R2dbcAutoConfiguration` is triggered by the presence of `META-INF/services/io.r2dbc.spi.ConnectionFactoryProvider` This SPI is included in r2dbc-proxy so we need to get rid of it. Maybe we should provided shaded version of the library instrumentation.